### ISSUE TITLE
Do not trigger `UnusedPrivateMember` for methods annotated with `OnLifecycleEvent`

### ISSUE THREAD

**Issue by igorwojda** [JLLeitschuh/ktlint-gradle#375]
Implement LifecycleObserver interface and annotating methods with `OnLifecycleEvent` is the new recommended Google approach allowing to react to lifecycle changes.

```
class Test : LifecycleObserver {
    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    fun onPause() {
        //...
    }
}
```

`UnusedPrivateMember` should take this into consideration (treat such methods as used)

More
https://developer.android.com/topic/libraries/architecture/lifecycle#lc

**Comment by igorwojda** [comment]
Should be detekt issue, sorry
